shared/src/main/scala/io/lemonlabs/uri/Uri.scala

```diff
@@ -953,6 +974,20 @@ final case class AbsoluteUrl(scheme: String,
 
   private[uri] def toString(c: UriConfig, hostToString: Host => String): String =
     scheme + "://" + authority.toString(c, hostToString) + path.toString(c) + queryToString(c) + fragmentToString(c)
+
+  def normalize(removeEmptyPathParts: Boolean = false, slashTerminated: Option[Boolean] = None): Self = {
+    val scheme = this.scheme.toLowerCase
+    val isHttp = scheme == "http" || scheme == "https"
+    copy(
+      scheme = scheme,
+      authority = authority.normalize(if (isHttp) Some(scheme) else None),
+      path = if (isHttp && path == EmptyPath) {
```

Comment:
I think we should remove the isHttp check here (more detail in https://github.com/lemonlabsuk/scala-uri/pull/325/files#r650263055)